agent/engine/docker_task_engine.go

```diff
@@ -245,6 +245,7 @@ func (engine *DockerTaskEngine) synchronizeState() {
 	}
 
 	tasks := engine.state.AllTasks()
+	var tasksToStart []*api.Task
 	for _, task := range tasks {
```

Comment:
please refactor contents of this `for` loop into a method while you're here. Same goes for https://github.com/aws/amazon-ecs-agent/pull/1036/files#diff-b9aa12c3425c34d6f330f28407f779f6R255